src/gui/updater/appimageupdater.cpp

```diff
@@ -0,0 +1,206 @@
+ */
+class AppImageUpdaterShim : public QObject
+{
+    Q_OBJECT
+
+private:
+    explicit AppImageUpdaterShim(const QString &zsyncFileUrl, QObject *parent = nullptr)
+        : QObject(parent)
+        , _updater(Utility::appImageLocation().toStdString(), true)
+    {
+        QString updateInformation(QStringLiteral("zsync|") + zsyncFileUrl);
+        _updater.setUpdateInformation(updateInformation.toStdString());
+    }
+
+    void _logStatusMessages()
+    {
+        std::string currentStatusMessage;
+
+        while (_updater.nextStatusMessage(currentStatusMessage)) {
+            qCInfo(lcUpdater) << "AppImageUpdate:" << QString::fromStdString(currentStatusMessage);
+        }
+    }
+
+public:
+    static AppImageUpdaterShim *makeInstance(const QString &updateInformation, QObject *parent)
+    {
+        try {
+            return new AppImageUpdaterShim(updateInformation, parent);
+        } catch (const std::exception &e) {
+            qCCritical(lcUpdater) << "Failed to create updater shim:" << e.what();
+            return nullptr;
+        }
+    }
+
+    bool isUpdateAvailable() noexcept
+    {
+        try {
+            bool updateAvailable;
+
+            if (!_updater.checkForChanges(updateAvailable)) {
+                _logStatusMessages();
+                return false;
+            }
+
+            _logStatusMessages();
+            return updateAvailable;
+        } catch (const std::exception &e) {
+            _logStatusMessages();
+            qCCritical(lcUpdater) << "Checking for update failed:" << e.what();
+            return false;
+        }
+    }
+
+    void startUpdateInBackground() noexcept
+    {
+        // monitor progress and log status messages
+        auto *timer = new QTimer(this);
+
+        timer->setInterval(100);
+
+        connect(timer, &QTimer::timeout, this, [=]() {
+            _logStatusMessages();
+
+            if (!_updater.isDone()) {
+                emit finished(!_updater.hasError());
+            }
+
+            timer->stop();
+        });
+
+        _updater.start();
+        timer->start();
+    }
+
+signals:
+    void finished(bool successfully);
+
+private:
+    appimage::update::Updater _updater;
+};
+
+} // namespace
+
+AppImageUpdater::AppImageUpdater(const QUrl &url)
+    : OCUpdater(url)
+{
+}
+
+
+bool AppImageUpdater::handleStartup()
+{
+    // nothing to do, update will be performed while app is running, if anything
+    return false;
+}
+
+void AppImageUpdater::versionInfoArrived(const UpdateInfo &info)
+{
+    const auto &currentVersion = Version::versionWithBuildNumber();
+    const auto newVersion = QVersionNumber::fromString(info.version());
+
+    if (info.version().isEmpty() || currentVersion >= newVersion) {
+        qCInfo(lcUpdater) << "Client is on latest version!";
+        setDownloadState(UpToDate);
+        return;
+    }
+
+    QSettings settings(ConfigFile::configFile(), QSettings::IniFormat);
```

Comment:
If we don't have a factory for that qsettings object, please add one.
Needing to pass the format looks weird.

Reply:
I'd rather do that in a second PR, and replace all occurrences, wdyt?